src/v2/view-builder/views/ov/ChallengeOktaVerifyResendPushView.js

```diff
@@ -18,41 +18,25 @@ const Body = BaseForm.extend(Object.assign(
-      if (messagesObjs && Array.isArray(messagesObjs.value)) {
-        this.add('<div class="ion-messages-container"></div>', '.o-form-error-container');
-
-        messagesObjs.value.forEach(messagesObj => {
-          const msg = messagesObj.message;
-          if (messagesObj?.class === 'ERROR') {
-            const options = {
-              content: msg,
-              type: 'error',
-            };
-            if (this.options.appState.containsMessageWithI18nKey(OV_NMC_FORCE_UPGRAGE_SERVER_KEY)) {
-              // add a title for OV force upgrade
-              options.title = loc('oie.numberchallenge.force.upgrade.title', 'login');
-            } else if (this.options.appState.containsMessageWithI18nKey(OV_UV_ENABLE_BIOMETRIC_SERVER_KEY)) {
-              // add a title for OV enable biometrics message during verification
-              options.content = null;
-              options.className = 'okta-verify-uv-callout-content';
-              options.title = loc('oie.authenticator.app.method.push.verify.enable.biometrics.title', 'login');
-              options.subtitle = loc('oie.authenticator.app.method.push.verify.enable.biometrics.description', 'login');
-              options.bullets = [
-                loc('oie.authenticator.app.method.push.verify.enable.biometrics.point1', 'login'),
-                loc('oie.authenticator.app.method.push.verify.enable.biometrics.point2', 'login'),
-                loc('oie.authenticator.app.method.push.verify.enable.biometrics.point3', 'login')
-              ];
-            }
-            this.add(createCallout(options), '.o-form-error-container');
-          } else {
-            this.add(`<p>${msg}</p>`, '.ion-messages-container');
-          }
-        });
-
+      let options = {};
+      if (this.options.appState.containsMessageWithI18nKey(OV_NMC_FORCE_UPGRAGE_SERVER_KEY)) {
+        // add a title for OV force upgrade
+        options.title = loc('oie.numberchallenge.force.upgrade.title', 'login');
+      } else if (this.options.appState.containsMessageWithI18nKey(OV_UV_ENABLE_BIOMETRIC_SERVER_KEY)) {
+        // add a title for OV enable biometrics message during verification
+        options.content = null;
+        options.className = 'okta-verify-uv-callout-content';
+        options.title = loc('oie.authenticator.app.method.push.verify.enable.biometrics.title', 'login');
+        options.subtitle = loc('oie.authenticator.app.method.push.verify.enable.biometrics.description', 'login');
+        options.type = 'error';
+        options.bullets = [
+          loc('oie.authenticator.app.method.push.verify.enable.biometrics.point1', 'login'),
+          loc('oie.authenticator.app.method.push.verify.enable.biometrics.point2', 'login'),
+          loc('oie.authenticator.app.method.push.verify.enable.biometrics.point3', 'login')
+        ];
+        options = createCallout(options);
       }
+      BaseForm.prototype.showMessages.call(this, options);
```

Comment:
Before, for each of the message we were evaluating `containsMessageWithI18nKey` and deciding what the content of the callout was
With this change we are passing `options` based on whatever the first message is. I don't think we will ever get the a messages object with these 2 errors together, but still this is prone to not render the right message. I made a comment in BaseForm that indicates where this is rendering the same view for all messages

Reply:
replied here https://github.com/okta/okta-signin-widget/pull/2281#discussion_r741309579